pkg/k8s/client.go

```diff
@@ -58,6 +58,18 @@ func NewClient(osmNamespace, meshConfigName string, informerCollection *osminfor
 		informerInitHandlerMap[informer]()
 	}
 
+	shouldObserve := func(obj interface{}) bool {
```

Comment:
let's remove these calls here, and add the the event handlers to the informer's `With*` methods

Reply:
Can you clarify the idea?

Reply:
I actually don't think these event handlers belong in the With* methods; this code seems to be a correct usage of AddEventHandler

Reply:
sorry i was mistaken ,its' actually in the k8s.client init* methods. take a look at PR https://github.com/openservicemesh/osm/pull/5006 or other instances in the k8/client.go file.